egs2/myst/asr1/local/data.sh

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+
+set -e
+set -u
+set -o pipefail
+
+log() {
+    local fname=${BASH_SOURCE[1]##*/}
+    echo -e "$(date '+%Y-%m-%dT%H:%M:%S') (${fname}:${BASH_LINENO[0]}:${FUNCNAME[1]}) $*"
+}
+SECONDS=0
+
+
+stage=1
+stop_stage=100000
+flac2wav=false
+nj=32
+
+log "$0 $*"
+. utils/parse_options.sh
+
+. ./db.sh
+. ./path.sh
+. ./cmd.sh
+
+
+
+if [ $# -ne 0 ]; then
+    log "Error: No positional arguments are required."
+    exit 2
+fi
+
+if [ -z "${MYST}" ]; then
+    log "Fill the value of 'MYST' of db.sh"
+    exit 1
+fi
+
+if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
+    if [ ! -d "${MYST}/myst_child_conv_speech" ]; then
+	    echo "stage 1: Download data to ${MYST}"
+        exit 1
+    else
+        log "stage 1: ${MYST}/myst_child_conv_speech is already existing. Skip data downloading"
+    fi
+fi
+
+if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ]; then
+    if "${flac2wav}"; then
```

Comment:
Is there a specific reason of introducing this stage?

Reply:
Here converting all .flac files to .wav files in the corpus can avoid sox backend error in stage 3. This benefits users who install sox via conda (current ESPnet installation does so by default) because sox provided by conda does not support flac audio format. If the user already have sox that support flac, thay can skip this stage by setting `local_data_opts=""`
and `audio_format=flac` in `run.sh` (see https://github.com/eric102004/espnet/blob/a59859c43560153a7e4a22a832c34bd85b7482ce/egs2/myst/asr1/run.sh#L21-L27).

Reply:
Thanks for the note. May I ask if there is a necessity in using sox? The default soundfile can handle flac format directly.

Reply:
Yeah, it looks like in stage 2 `scripts/utils/perturb_data_dir_speed.sh` uses Sox to perturb the speed, so we cannot bypass Sox without modifying this script.

Reply:
One another dataset including flac files is LibriSpeech. It uses `flac` along with Sox to perturb the audio. The difference is  that they did it on-the-fly. See https://github.com/espnet/espnet/blob/0eeec05b03d0ccba7a102360219b767d0773c518/egs/librispeech/asr1/local/data_prep.sh#L56